xarray/core/utils.py

```diff
@@ -496,6 +496,7 @@ def _warn(self) -> None:
             "in order to be more consistent with `DataArray.dims`. To access a mapping from dimension names to lengths, "
             "please use `Dataset.sizes`.",
             FutureWarning,
+            stacklevel=3,
```

Comment:
mind switching it to https://github.com/pydata/xarray/blob/9acc411bc7e99e61269eadf77e96b9ddd40aec9e/xarray/core/utils.py#L1256 please?

Reply:
sure. i mean, i feel like stack inspection is so overkill though. the call the warning is mean to be fast.  any kind of stack inspection in my mind is going to be slower.

The other advantage of emitting stacklevel=3 is that you can actually find internal instances of this within xarray.

However, as you wish. Please feel free to modify this as you see fit and merge, it is really meant to be a simple one liner and merged with ease.

Reply:
Yes, fair — it is definitely a perf tradeoff. 

"Ideally" the warnings aren't firing in hot paths / folks rework their code to prevent them. But it is a bit fuzzy. 

If we can find a case where it doesn't work well, we could def change the policy...